File: 01_Programming_Basics/01_Programming_Basics_with_C++/08_While_Loop_Lab/05_Walking.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
  string input;
  int steps = 0;
  int totalSteps = 0;

  getline(cin, input);

  while(true) {
    if (input == "Going home") {
      getline(cin, input);
      steps = stoi(input);
      totalSteps += steps;
      break;
    }
    
    steps = stoi(input);
    totalSteps += steps;

    if (totalSteps >= 10000) { break; }
    
    getline(cin, input);
  }

  if (totalSteps >= 10000 ) {
    cout << "Goal reached! Good job!" << endl;
  } else {
    cout << 10000 - totalSteps << " more steps to reach goal." << endl;
  }

  return 0;

}

File: 01_Programming_Basics/03_Programming_Basics_with_JavaScript/05_While_Loop/05_Walking.js
function Walking(args) {
  let i = 0;
  let command = args[i];
  let steps = 0;

  while(steps < 10000) {

    if (command == "Going home") {
      steps += parseInt(args[i + 1]);
      break;
    }

    console.log("loop " + i);
    console.log(command);
    steps += parseInt(command);

    i++;
    command = args[i];
  }

console.log(steps);

  if (steps < 10000) {
    console.log(10000 - steps + " more steps to reach goal.");
  } else {  
    console.log("Goal reached! Good job!");
  }
}